client/src/use/useSettings.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  reactive, ref,
+} from '@vue/composition-api';
+
+export interface NewTrackSettings {
+    mode: string;
+    type: string;
+    modeSettings: {
+      Track: {
+        autoAdvanceFrame: boolean;
+      };
+      Detection: {
+        continuous: boolean;
+      };
+    };
+  }
+export interface ClientSettings {
+  NewTrackSettings: NewTrackSettings;
+}
+
+export default function useSettings() {
+  const clientSettings = reactive({
+    newTrackSettings: {
+      mode: ref('Track'),
```

Comment:
reactive() and refs()  are not meant to be used together.  if you want refs on export, use `toRefs`.  Then `newTrackSettings` will be one giant ref, and if you destructure it, you'll break reactivity.  

That's probably what I'd recommend.

Reply:
I was hoping to avoid utilizing the `.value` notation for stuff.  I don't destructure it but I'm guessing that will probably break the reactivity.  I so hoping for some cleaner access to data.
![image](https://user-images.githubusercontent.com/61746913/85780002-29550500-b6f2-11ea-92f2-47802da3ca82.png)


They add the caveat to this further down the page about the reference destructuring when dealing with a reactive component.

Reply:
[toRefs](https://github.com/vuejs/composition-api/blob/233dafaf804b5b4d4876fd373c6f2dbf405b0d2a/src/reactivity/ref.ts#L116
)
So it is just doing a shallow traverse and applying refs(property) to each item.  So I'm guessing that what I did before was a more targeted version of this?  By making the specific children of the object Refs they avoid the issue of destructuring causing them to loose their reactivity and can be referenced without the value inside of templates.  I don't know for certain though.  The weird issue is the sub objects.